app/models/location_deactivation_period.rb

```diff
@@ -24,6 +26,8 @@ def validate_reactivation(record, recent_deactivation, location)
   end
 
   def validate_deactivation(record, location)
+    earliest_possible_deactivation = FormHandler.instance.in_crossover_period? ? previous_collection_start_date : current_collection_start_date
```

Comment:
We're duplicating this logic four times. Could this live on the form handler itself? What would be an appropriate name to describe it?